packages/web3-providers-base/test/unit/providerString.test.ts

```diff
@@ -0,0 +1,19 @@
+import Web3ProviderBase from '../../src/index'
+
+describe('Web3ProviderBase.providerString', () => {
+    const providerOptions = {
+        providerString: 'http://127.0.0.1:8545'
+    }
+    const updatedProviderString = 'teststring'
+
+    it('should return providerString', () => {
+        const web3ProviderBase = new Web3ProviderBase(providerOptions)
+        expect(web3ProviderBase.providerString).toBe(providerOptions.providerString)
+    })
+
+    it('should update providerString', () => {
+        const web3ProviderBase = new Web3ProviderBase(providerOptions)
+        web3ProviderBase.providerString = updatedProviderString
+        expect(web3ProviderBase.providerString).toBe(updatedProviderString)
+    })
```

Comment:
Add a test case to check if it will throw when you query for `web3Prodiver._providerString` its a sanity, but ensures no one removes the `private` from it in the future

Reply:
I tried, but tsc won't compile it :(

Reply:
oh funny lol. That makes sense becasue its not allowed ahhaha